### DISCUSSION TITLE
[Question] Is there a reason to consider using WSL?

### DISCUSSION THREAD

**Discussion by myndxero** [vladmandic/sdnext#1232] [Q&A]
I was kinda dipping my toes reading about directml and voltaml and stuff like that. Seems WSL is used more than trying to anything natively in Windows. I'm not a linux guy, I have my steam deck and I've done some casual-level heavy stuff in proton, but that's the extent of my present experiences with linux.

Just wondering if, with emerging AI technologies, it would be worth my time to get comfortable with WSL at least. If there's any real benefit right now. I've seen VLAD state firmly there's not really a performance difference between the two, and that's not necessarily what I am looking for with using WSL, I think it's just more along the lines of accessibility. Choosing to be at the mercy of those willing to port/guide us Windows plebs when they have the time/inclination, or not.

Thanks.

**Comment by vladmandic** [comment]
WSL has no downsides for backends which fully support virtualization (CUDA, DirectML). ROCm doesn't and IPEX does partially.
But if you're just a user, there are no significant benefits either. Only issues that are Windows specific are sometimes (rarely) PIP messes things up during package upgrades since it (still) doesn't handle temporary files nicely so it can run into file-locking access violation and (worst-case) corrupt `venv`.
I use WSL as my primary dev system purely because dev experience under Linux is much cleaner. But that's dev, that's not user. So no, its not worth learning WSL if you don't need it.
